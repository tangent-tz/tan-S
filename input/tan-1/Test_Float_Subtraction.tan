main {
    const floatone := 10.5;
    const floattwo := 3.25;

    # Basic subtraction
    const resultone := floatone - floattwo;
    print resultone;  # Output: 7.25
    print \n;

    # Subtraction with negative numbers
    const floatthree := -5.5;
    const floatfour := -2.75;
    const resulttwo := floatthree - floatfour;
    print resulttwo;  # Output: -2.75
    print \n;

    # Subtraction with mixed positive and negative numbers
    const floatfive := 8.75;
    const floatsix := -3.5;
    const resultthree := floatfive - floatsix;
    print resultthree;  # Output: 12.25
    print \n;

    # Subtraction resulting in zero
    const floatseven := 10.0;
    const floateight := 10.0;
    const resultfour := floatseven - floateight;
    print resultfour;  # Output: 0.0
    print \n;

    # Subtraction with large numbers
    const floatnine := 999.99;
    const floatten := 500.0;
    const resultfive := floatnine - floatten;
    print resultfive;  # Output: 499.99
    print \n;
}
