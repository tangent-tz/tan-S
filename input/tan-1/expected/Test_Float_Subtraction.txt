7.250000
-2.750000
12.250000
0.000000
499.990000